doc/modules/preprocessing.rst

```diff
@@ -830,6 +830,72 @@ lexicon order.
    >>> enc.infrequent_categories_
    [array(['b', 'c'], dtype=object)]
 
+.. _target_encoder:
+
+Target Regressor Encoder
+------------------------
+
+.. currentmodule:: sklearn.preprocessing
+
+The :class:`TargetEncoder` uses target mean groupby the
+categorical feature for encoding the categories [PAR]_ [MIC]_. This encoding
+scheme is useful with categorical features with high cardinality, where one hot
+encoding would inflate the feature space making it more expensive for a
+downstream model to process. A classical example of high cardinality categories
+are location based such as zip code or region. For the binary classification
```

Comment:
```suggestion
are location based such as zip code or region. For a binary classification
```
so is this doing just regression or regression and binary classification for now?